extensions/amp-springboard-player/0.1/amp-springboard-player.js

```diff
@@ -0,0 +1,139 @@
+
+class AmpSpringboardPlayer extends AMP.BaseElement {
+
+  /** @override */
+  createdCallback() {
+    this.preconnect.url('https://cms.springboardplatform.com');
+    this.preconnect.url('https://www.springboardplatform.com');
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    const width = this.element.getAttribute('width');
+    const height = this.element.getAttribute('height');
+    const mode = AMP.assert(
+        this.element.getAttribute('data-mode'),
+        'The data-mode attribute is required for <amp-springboard-player> %s',
+        this.element);
+    const contentId = AMP.assert(
+        this.element.getAttribute('data-content-id'),
+        'The data-content-id attribute is required for' +
+        '<amp-springboard-player> %s',
+        this.element);
+    const domain = AMP.assert(
+        this.element.getAttribute('data-domain'),
+        'The data-domain attribute is required for <amp-springboard-player> %s',
+        this.element);
+
+    this.width_ = width;
+    this.height_ = height;
+    this.mode_ = mode;
+    this.contentId_ = contentId;
+    this.domain_ = domain;
+
+    if (!this.getPlaceholder()) {
+      this.buildImagePlaceholder_();
+    }
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = document.createElement('iframe');
+    const siteId = AMP.assert(
+        this.element.getAttribute('data-site-id'),
+        'The data-site-id attribute is required for' +
+        '<amp-springboard-player> %s',
+        this.element);
+    const playerId = AMP.assert(
+        this.element.getAttribute('data-player-id'),
+        'The data-player-id attribute is required for' +
+        '<amp-springboard-player> %s',
+        this.element);
+    const items = this.element.getAttribute('data-items') || '10';
+
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
+    iframe.id = playerId + '_' + this.contentId_;
+    iframe.src = 'https://cms.springboardplatform.com/embed_iframe/' +
+    	encodeURIComponent(siteId) + '/' + encodeURIComponent(this.mode_) +
+    	'/' + encodeURIComponent(this.contentId_) + '/' +
+    	encodeURIComponent(playerId) + '/' + encodeURIComponent(this.domain_) +
+    	'/' + encodeURIComponent(items);
+    this.applyFillContent(iframe);
+    iframe.width = this.width_;
+    iframe.height = this.height_;
+    /** @private {?Element} */
+    this.iframe_ = iframe;
+    this.element.appendChild(iframe);
+    return loadPromise(iframe);
+  }
+
+  /** @override */
+  documentInactiveCallback() {
+    if (this.iframe_ && this.iframe_.contentWindow) {
+      this.iframe_.contentWindow./*OK*/postMessage('ampPause', '*');
+    }
+    return false;
+  }
+
+  /** @private */
+  buildImagePlaceholder_() {
+    const imgPlaceholder = new Image();
+
+    setStyles(imgPlaceholder, {
+      //Cover matches YouTube Player styling.
```

Comment:
Fix the comments referring to YouTube. 

And make sure the styling actually matches the springboard player - this means when the placeholder image gets hidden it needs to be the exact poster image of the video rather than having different styling - you can test this by throttling the network in chrome devtools and loading the example and noticing that the image placeholder first loads and then gets replaced by the video. 

Just make sure the styling doesn't differ too much.